lib/chef/knife/bootstrap/options.rb

```diff
@@ -0,0 +1,356 @@
+        # TODO this should be available from train, which we can make our source of truth.
+        # TODO - we don't actually validate that the protocol is valid...
+        WINRM_AUTH_PROTOCOL_LIST = %w{plaintext kerberos ssl negotiate}.freeze
+
+        # TODO - missing - authtimeout (minutes)
+        # TODO - missing impl - session-timeout minutes
+        def self.included(includer)
+          includer.class_eval do
+
+            # Common connectivity options
+            option :connection_user,
+              short: "-U USERNAME",
+              long: "--connection-user USERNAME",
+              description: "Authenticate to the target host with this user account"
+
+            option :password,
+              short: "-P PASSWORD",
+              long: "--connection-password PASSWORD",
+              description: "Authenticate to the target host with this password"
+
+            option :connection_port,
+              short: "-p PORT",
+              long: "--connection-port PORT",
+              description: "The port on the target node to connect to."
+
+            option :connection_protocol,
+              short: "-o PROTOCOL",
+              long: "--connection-protocol PROTOCOL",
+              description: "The protocol to use to connect to the target node.  Supports ssh and winrm."
+
+            option :max_wait,
+              short: "-W SECONDS",
+              long: "--max-wait SECONDS",
+              description: "The maximum time to wait for the initial connection to be established."
+
+            ## SSH options
+
+            option :ssh_gateway,
+              short: "-G GATEWAY",
+              long: "--ssh-gateway GATEWAY",
+              description: "The ssh gateway",
+              proc: Proc.new { |key| Chef::Config[:knife][:ssh_gateway] = key }
+
+            option :ssh_gateway_identity,
+              long: "--ssh-gateway-identity SSH_GATEWAY_IDENTITY",
+              description: "The SSH identity file used for gateway authentication",
+              proc: Proc.new { |key| Chef::Config[:knife][:ssh_gateway_identity] = key }
+
+            # SSH train ssh: options[:forward_agent]
+            option :ssh_forward_agent,
+              short: "-A",
+              long: "--ssh-forward-agent",
+              description: "Enable SSH agent forwarding",
+              boolean: true
+
+            # SSH train: options[key_files]
+            option :ssh_identity_file,
+              short: "-i IDENTITY_FILE",
+              long: "--ssh-identity-file IDENTITY_FILE",
+              description: "The SSH identity file used for authentication"
+
+            # ssh options - train options[:verify_host_key]
+            option :ssh_verify_host_key,
+              long: "--ssh-[no-]verify-host-key",
+              description: "Verify host key, enabled by default.",
+              boolean: true
+
+            # argument to installer in chef-full, via bootstrap_context
+            option :prerelease,
```

Comment:
We should have a discussion at some point if we need this option. We don't release pre-release gems anymore so it's basically a useless option.

Reply:
Removed after discussion in community slack